css/src/components/timeline.scss

```diff
@@ -0,0 +1,74 @@
+@use 'sass:math';
+$timeline-content-font-size: $font-size-8 !default;
+$timeline-timestamp-font-size: $font-size-9 !default;
+$timeline-timestamp-font-weight: $weight-semilight !default;
+$timeline-description-padding-top: $spacer-4 !default;
+$timeline-description-line-height: $line-height-normal;
+
+$timeline-item-border-left: $border-width solid $border !default;
+$timeline-item-min-height: $layout-6 !default;
+$timeline-item-padding-bottom: $spacer-7 !default;
+$timeline-item-padding-left: $spacer-8 !default;
+
+$timeline-item-template-gap: 8px !default;
+$timeline-item-template-width: 100% !default;
+
+$timeline-item-badge-transform: translate(
```

Comment:
This transform is causing issues when the `dir="rtl"` is applied to the page:

![Timeline in RTL view. The badge is too far left, overlapping the timeline item text.](https://github.com/user-attachments/assets/6becc20a-d002-44df-9c45-4a9423da7075)


I suggest maybe having `$timeline-item-badge-transform-ltr` and `$timeline-item-badge-transform-rtl` variables, and using `:dir(rtl)` to specify the RTL one.

Reply:
Great fine, perhaps this?

```scss


$timeline-item-badge-transform: translate(
	calc(-#{$timeline-item-padding-left} - 50%),
	-#{math.div($timeline-timestamp-font-size, 2)}
);

$timeline-item-badge-transform-rtl: translate(
	calc(#{$timeline-item-padding-left} + 50%),
	-#{math.div($timeline-timestamp-font-size, 2)}
);

.timeline {
	font-size: $document-font-size;
	container-type: inline-size;

	.timeline-item {
		display: flex;
		position: relative;
		padding-block-end: $timeline-item-padding-bottom;
		padding-inline-start: $timeline-item-padding-left;
		border-inline-start: $timeline-item-border-left;

		&:last-child {
			border-inline-start: none;
		}

		.timeline-item-badge {
			position: absolute;
			transform: $timeline-item-badge-transform;

			[dir='rtl'] & {
				transform: $timeline-item-badge-transform-rtl;
			}
		}

Reply:
I wouldn't recommend `[dir='rtl'] &`, because it can be foiled by markup like:

```html
<html dir="rtl">
    <body dir="ltr">
        <!-- Timeline would adopt RTL rules, despite surrounding contents being LTR -->
```
We apply `dir` at a few levels in our markup as is, and while I'm pretty sure they're all kept in sync, using `&:dir(rtl)` would guarantee the rules match the direction at the same level as the timeline component.

Reply:
good catch, updated to have rtl

Reply:
Thanks @BenDMyers, your selector is correct! Thanks

Reply:
Looks much better now!

![RTL timeline component, with badge properly aligned to the right.](https://github.com/user-attachments/assets/5dd1a52b-b99d-4620-96fb-e82810a550e7)